src/WindowsAzureMessaging/WindowsAzureMessaging/Models/MSInstallation.m

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSInstallation.h"
+
+@implementation MSInstallation
+
+- (void)encodeWithCoder:(nonnull NSCoder *)coder {
+    [coder encodeObject:self.installationID forKey:@"installationID"];
+    [coder encodeObject:self.pushChannel forKey:@"pushChannel"];
+    [coder encodeObject:self.platform forKey:@"platform"];
+    [coder encodeObject:self.tags forKey:@"tags"];
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+    if (self = [super init]) {
+        self.installationID = [coder decodeObjectForKey:@"installationID"] ?: [[NSUUID UUID] UUIDString];
+        self.pushChannel = [coder decodeObjectForKey:@"pushChannel"];
+        self.platform = [coder decodeObjectForKey:@"platform"] ?: @"APNS";
+        self.tags = [coder decodeObjectForKey:@"tags"];
+    }
+    
+    return self;
+}
+
+- (instancetype) init {
+    if(self = [super init]) {
+        self.installationID = [[NSUUID UUID] UUIDString];
+        self.platform = @"APNS";
+    }
+    
+    return self;
+}
+
+- (instancetype) initWithDeviceToken:(NSString *) deviceToken {
+    if (self = [self init]) {
+        self.pushChannel = deviceToken;
+    }
+    
+    return self;
+}
+
++ (MSInstallation *) createFromDeviceToken:(NSString *) deviceToken {
+    return [[MSInstallation alloc] initWithDeviceToken:deviceToken];
+}
+
++ (MSInstallation *) createFromJsonString:(NSString *)jsonString {
+    MSInstallation *installation = [MSInstallation new];
+    NSData * data = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+    
+    NSError *error = nil;
+    NSDictionary *dictionary = [NSJSONSerialization
+                                 JSONObjectWithData:data
+                                 options:0
+                                 error:&error];
+    
+    installation.installationID = dictionary[@"installationId"];
+    installation.platform = dictionary[@"platform"];
+    installation.pushChannel = dictionary[@"pushChannel"];
+    installation.tags = dictionary[@"tags"];
+    
+    return installation;
+}
+
+- (NSData *) toJsonData {
+    
+    NSDictionary * dictionary = @{
+       @"installationId" : self.installationID,
+       @"platform" : self.platform,
+       @"pushChannel" : self.pushChannel,
+       @"tags" : self.tags ?: @""
```

Comment:
If `tags` is not set, the default value should be an empty array, not an empty string.

Reply:
Fixed